src/init_macro.rs

```diff
@@ -235,6 +249,28 @@ pub mod signal_watcher {
     }
 }
 
+#[derive(NativeClass, Default)]
+#[inherit(Node)]
+struct InputEventWatcher {
+    notification_channel: Option<Sender<Ref<InputEvent>>>,
+}
+
+#[methods]
+impl InputEventWatcher {
+    fn new(_base: &Node) -> Self {
+        Self::default()
+    }
+
+    #[export]
+    fn _unhandled_input(&mut self, _base: TRef<Node>, input_event: Ref<InputEvent>) {
```

Comment:
Not sure what input functions we want to handle here. My impression was that `_unhandled_input` is the most commonly used, but maybe it should allow for `_input` and `_unhandled_key_input` as well.

Also in the future we may want to classify the actual `InputEvent`s and emit them as separate Bevy events, so that we're not clogging the Bevy event system with e.g. `InputEventMouseMotion` every bevy loop, but maybe the overhead is not a concern - I'm not familiar here.

Reply:
I think there is a use-case for both handled and unhandled input events. My plan is to rewrite this listener class to emit a input event signal and use our Godot signal support. I believe we can send a high volume of events but I'll see.